core/src/parser.rs

```diff
@@ -78,7 +78,7 @@ pub enum ParseError {
 }
 
 /// Parse the given Rust source string into `ParsedData`.
-pub fn parse(input: &str) -> Result<ParsedData, ParseError> {
```

Comment:
`supports_flatten` shouldn't be something the parser needs to worry about; it should unconditionally emit the fully parsed data (including `#[serde(flatten)]` attributes), and then that parsed data can be rejected by individual languages when the output is generated, if necessary.

Reply:
Sounds fair, but concretely just wondering how you would see it implemented.